File: eclipse/FizzBuzz/src/de/welcz/Rule.java
package de.welcz;

class Rule {
	private final int divisor;
	private final String result;

	public Rule(int divisor, String result) {
		this.divisor = divisor;
		this.result = result;
	}

	static RuleSelector divisibleBy(int divisor) {
		return new RuleSelector(divisor);
	}

	boolean appliesTo(int number) {
		return number % divisor == 0;
	}

	String giveResult() {
		return result;
	}
}

File: typescript/fizzbuzz-mocha/src/rule.ts
export class Rule {
    get result(): string {
        return this._result;
    }

    private _result: string;
    private _denominators: number[];

    constructor(result: string, denominators: number[]) {
        this._result = result;
        this._denominators = denominators;
    }

    public appliesTo(input: number): boolean {
        return this._denominators.every((d) => input % d === 0);
    }
}
